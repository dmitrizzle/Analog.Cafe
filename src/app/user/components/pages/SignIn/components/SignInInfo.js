import React from "react"
import styled from "styled-components"

export const AccountBenifits = styled.ul`
  max-width: 19em;
  margin: 1.5em auto 0 !important;
  padding: 0.5em;
  background: #2c2c2c;
  color: #fff;
  /* border-radius: 0.25em; */
  /* box-shadow: 0 1px 1px rgba(44,44,44,0.25), 0 0 0 1px rgba(44,44,44,0.125); */
  font-size: 0.85em;
  font-style: italic;
  li {
    list-style-type: none;
    margin-bottom: 0.25em;
    padding: 0 !important;
    .icon {
      font-style: normal;
    }
    .highlight {
      color: ${props => props.theme.color.brand()};
    }
  }
`

export const Hint = props => (
  <p style={{ textAlign: "center", marginBottom: "0" }}>
    <small>
      {props.stateSessionInfo.hasLoggedIn &&
      props.stateSessionInfo.loginMethod ? (
        <em>
          Hint: last time you used {props.stateSessionInfo.loginEmail}{" "}
          {props.stateSessionInfo.loginMethod}.
        </em>
      ) : (
        <span>&nbsp;</span>
      )}
    </small>
  </p>
)

export default props => {
  return (
    <React.Fragment>
      <AccountBenifits>
        <li>
          <span className="highlight icon">❤</span> Vote for and save your
          favourites.
        </li>
        <li>
          <span className="icon">✒︎</span> Submit your work to get featured.
        </li>
        <li>
          <span className="icon">❖</span> Free printables, exclusives & email
          updates.
        </li>
      </AccountBenifits>
    </React.Fragment>
  )
}
