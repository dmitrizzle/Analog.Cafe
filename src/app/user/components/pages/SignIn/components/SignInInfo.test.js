import { shallow } from "enzyme"
import React from "react"

import SignInInfo from "./SignInInfo"

it("Render SignInInfo without crashing", () => {
  shallow(<SignInInfo stateSessionInfo={{}} />)
})
