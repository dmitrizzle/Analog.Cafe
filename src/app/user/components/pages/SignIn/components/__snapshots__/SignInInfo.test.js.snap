// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SignInInfo, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    stateSessionInfo={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <styled.ul>
        <li>
                <span
                        className="highlight icon"
                >
                        ❤
                </span>
                 Save your
                 
                <Unknown
                        to="/favourites"
                >
                        favourites
                </Unknown>
                 for later.
        </li>
        <li>
                <span
                        className="icon"
                >
                        ✒︎
                </span>
                 
                <Unknown
                        to="/submit"
                >
                        Submit
                </Unknown>
                 your work, get featured.
        </li>
        <li>
                <span
                        className="highlight icon"
                >
                        ❖
                </span>
                 Free
                 
                <Unknown
                        to="/resources"
                >
                        downloads
                </Unknown>
                 & pre-releases.
        </li>
        <li>
                <span
                        className="icon"
                >
                        ☞
                </span>
                 Weekly email
                 
                <Unknown
                        onClick={[Function]}
                        to="#analogue-reads"
                >
                        newsletter
                </Unknown>
                .
        </li>
</styled.ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <li>
            <span
                        className="highlight icon"
            >
                        ❤
            </span>
             Save your
             
            <Unknown
                        to="/favourites"
            >
                        favourites
            </Unknown>
             for later.
</li>,
          <li>
            <span
                        className="icon"
            >
                        ✒︎
            </span>
             
            <Unknown
                        to="/submit"
            >
                        Submit
            </Unknown>
             your work, get featured.
</li>,
          <li>
            <span
                        className="highlight icon"
            >
                        ❖
            </span>
             Free
             
            <Unknown
                        to="/resources"
            >
                        downloads
            </Unknown>
             & pre-releases.
</li>,
          <li>
            <span
                        className="icon"
            >
                        ☞
            </span>
             Weekly email
             
            <Unknown
                        onClick={[Function]}
                        to="#analogue-reads"
            >
                        newsletter
            </Unknown>
            .
</li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="highlight icon"
>
                ❤
</span>,
              " Save your",
              " ",
              <Unknown
                to="/favourites"
>
                favourites
</Unknown>,
              " for later.",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "❤",
                "className": "highlight icon",
              },
              "ref": null,
              "rendered": "❤",
              "type": "span",
            },
            " Save your",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "favourites",
                "to": "/favourites",
              },
              "ref": null,
              "rendered": "favourites",
              "type": [Function],
            },
            " for later.",
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="icon"
>
                ✒︎
</span>,
              " ",
              <Unknown
                to="/submit"
>
                Submit
</Unknown>,
              " your work, get featured.",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "✒︎",
                "className": "icon",
              },
              "ref": null,
              "rendered": "✒︎",
              "type": "span",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Submit",
                "to": "/submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": [Function],
            },
            " your work, get featured.",
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="highlight icon"
>
                ❖
</span>,
              " Free",
              " ",
              <Unknown
                to="/resources"
>
                downloads
</Unknown>,
              " & pre-releases.",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "❖",
                "className": "highlight icon",
              },
              "ref": null,
              "rendered": "❖",
              "type": "span",
            },
            " Free",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "downloads",
                "to": "/resources",
              },
              "ref": null,
              "rendered": "downloads",
              "type": [Function],
            },
            " & pre-releases.",
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="icon"
>
                ☞
</span>,
              " Weekly email",
              " ",
              <Unknown
                onClick={[Function]}
                to="#analogue-reads"
>
                newsletter
</Unknown>,
              ".",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "☞",
                "className": "icon",
              },
              "ref": null,
              "rendered": "☞",
              "type": "span",
            },
            " Weekly email",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "newsletter",
                "onClick": [Function],
                "to": "#analogue-reads",
              },
              "ref": null,
              "rendered": "newsletter",
              "type": [Function],
            },
            ".",
          ],
          "type": "li",
        },
      ],
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <styled.ul>
          <li>
                    <span
                              className="highlight icon"
                    >
                              ❤
                    </span>
                     Save your
                     
                    <Unknown
                              to="/favourites"
                    >
                              favourites
                    </Unknown>
                     for later.
          </li>
          <li>
                    <span
                              className="icon"
                    >
                              ✒︎
                    </span>
                     
                    <Unknown
                              to="/submit"
                    >
                              Submit
                    </Unknown>
                     your work, get featured.
          </li>
          <li>
                    <span
                              className="highlight icon"
                    >
                              ❖
                    </span>
                     Free
                     
                    <Unknown
                              to="/resources"
                    >
                              downloads
                    </Unknown>
                     & pre-releases.
          </li>
          <li>
                    <span
                              className="icon"
                    >
                              ☞
                    </span>
                     Weekly email
                     
                    <Unknown
                              onClick={[Function]}
                              to="#analogue-reads"
                    >
                              newsletter
                    </Unknown>
                    .
          </li>
</styled.ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <li>
              <span
                            className="highlight icon"
              >
                            ❤
              </span>
               Save your
               
              <Unknown
                            to="/favourites"
              >
                            favourites
              </Unknown>
               for later.
</li>,
            <li>
              <span
                            className="icon"
              >
                            ✒︎
              </span>
               
              <Unknown
                            to="/submit"
              >
                            Submit
              </Unknown>
               your work, get featured.
</li>,
            <li>
              <span
                            className="highlight icon"
              >
                            ❖
              </span>
               Free
               
              <Unknown
                            to="/resources"
              >
                            downloads
              </Unknown>
               & pre-releases.
</li>,
            <li>
              <span
                            className="icon"
              >
                            ☞
              </span>
               Weekly email
               
              <Unknown
                            onClick={[Function]}
                            to="#analogue-reads"
              >
                            newsletter
              </Unknown>
              .
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="highlight icon"
>
                  ❤
</span>,
                " Save your",
                " ",
                <Unknown
                  to="/favourites"
>
                  favourites
</Unknown>,
                " for later.",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "❤",
                  "className": "highlight icon",
                },
                "ref": null,
                "rendered": "❤",
                "type": "span",
              },
              " Save your",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "favourites",
                  "to": "/favourites",
                },
                "ref": null,
                "rendered": "favourites",
                "type": [Function],
              },
              " for later.",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="icon"
>
                  ✒︎
</span>,
                " ",
                <Unknown
                  to="/submit"
>
                  Submit
</Unknown>,
                " your work, get featured.",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "✒︎",
                  "className": "icon",
                },
                "ref": null,
                "rendered": "✒︎",
                "type": "span",
              },
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Submit",
                  "to": "/submit",
                },
                "ref": null,
                "rendered": "Submit",
                "type": [Function],
              },
              " your work, get featured.",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="highlight icon"
>
                  ❖
</span>,
                " Free",
                " ",
                <Unknown
                  to="/resources"
>
                  downloads
</Unknown>,
                " & pre-releases.",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "❖",
                  "className": "highlight icon",
                },
                "ref": null,
                "rendered": "❖",
                "type": "span",
              },
              " Free",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "downloads",
                  "to": "/resources",
                },
                "ref": null,
                "rendered": "downloads",
                "type": [Function],
              },
              " & pre-releases.",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="icon"
>
                  ☞
</span>,
                " Weekly email",
                " ",
                <Unknown
                  onClick={[Function]}
                  to="#analogue-reads"
>
                  newsletter
</Unknown>,
                ".",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "☞",
                  "className": "icon",
                },
                "ref": null,
                "rendered": "☞",
                "type": "span",
              },
              " Weekly email",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "newsletter",
                  "onClick": [Function],
                  "to": "#analogue-reads",
                },
                "ref": null,
                "rendered": "newsletter",
                "type": [Function],
              },
              ".",
            ],
            "type": "li",
          },
        ],
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
